src/daft-plan/src/physical_planner/translate.rs

```diff
@@ -656,7 +659,7 @@ pub(super) fn translate_single_logical_node(
                     {
                         let split_op = PhysicalPlan::FanoutByHash(FanoutByHash::new(
                             right_physical,
-                            num_partitions,
+                            min(num_partitions, cfg.shuffle_join_default_partitions),
```

Comment:
I think we should have this be part of the initial `num_partitions` calculation, because it's possible that only one side gets repartitioned, and therefore possible that the resulting num partitions will be unequal. 

Example query:
```
df1 = df.repartition(100, 'col')
df2 = df.repartition(300, 'col')
df = df1.join(df2, on = 'col')
```
Here, both df1 and df2 are partitioned by hash before the join, respectively into 100 and 300. Therefore, we take the max of both to set `num_partitions = 300`, and only repartition the left side. However, `shuffle_join_default_partitions=200`, so we repartition to 200 instead. Now we have imbalanced partitions for the join.